src/senaite/sync/browser/add.py

```diff
@@ -0,0 +1,125 @@
+
+from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
+from plone import protect
+from senaite import api
+from senaite.sync import _
+from senaite.sync.browser.interfaces import ISync
+from senaite.sync.browser.views import Sync
+from senaite.sync.fetchstep import FetchStep
+from zope.interface import implements
+
+SYNC_STORAGE = "senaite.sync"
+PREFIX_SPECIAL_CHARACTERS = u"*.!$%&/()=-+:'`´^"
+
+class Add(Sync):
+    """ Add new sync instance view
+    """
+    implements(ISync)
+
+    template = ViewPageTemplateFile("templates/add.pt")
+
+    def __init__(self, context, request):
+        super(Sync, self).__init__(context, request)
+
+    def __call__(self):
+        protect.CheckAuthenticator(self.request.form)
+
+        self.portal = api.get_portal()
+        self.request.set('disable_plone.rightcolumn', 1)
+        self.request.set('disable_border', 1)
+
+        # Handle form submit
+        form = self.request.form
+        fetchform = form.get("fetchform", False)
+        dataform = form.get("dataform", False)
+        if not any([fetchform, dataform]):
+            return self.template()
+
+        # Handle "Fetch" action
+        if form.get("fetch", False):
+
+            url = form.get("url", "")
+            if not url.startswith("http"):
+                url = "http://{}".format(url)
+            domain_name = form.get("domain_name", None)
+            username = form.get("ac_name", None)
+            password = form.get("ac_password", None)
+            # check if all mandatory fields have values
+            if not all([domain_name, url, username, password]):
+                message = _("Please fill in all required fields")
+                self.add_status_message(message, "error")
+                return self.template()
+
+            import_settings = True if form.get("import_settings") == 'on' else False
+            import_users = True if form.get("import_users") == 'on' else False
+            import_registry = True if form.get("import_registry") == 'on' else False
+            content_types = form.get("content_types", None)
+            unwanted_content_types = form.get("unwanted_content_types", None)
+            prefix = form.get("prefix", None)
+            prefixable_types = form.get("prefixable_types", None)
+
+            # Content Type Validation
+            portal_types = api.get_tool("portal_types")
+            if content_types:
+                content_types = [t.strip() for t in content_types.split(",")]
+                content_types = filter(lambda ct: ct in portal_types,
+                                       content_types)
+
+            if unwanted_content_types:
+                unwanted_content_types = [t.strip() for t
+                                          in unwanted_content_types.split(",")]
+                unwanted_content_types = filter(lambda ct: ct in portal_types,
+                                                unwanted_content_types)
```

Comment:
I suggest to reduce a bit the complexity by adding a function to get the content types:
```python
def get_content_types(self, content_types):
    if not content_types:
        return list()
    portal_types = api.get_tool("portal_types")
    ...
    return content_types
```
So, you'll only need to do the following here:
```python
prefixable_types = self.get_content_types(form.get("prefixable_types", None))
content_types = self.get_content_types(form.get("content_types", None))
unwanted_content_types = self.get_content_types(format.get("unwanted_content_types", None))
```
Note with this change you'll be able to remove the logic for `prefixable_types` below.